.travis.yml

```diff
@@ -14,45 +14,52 @@ matrix:
             - cmake3
             - cmake3-data
             - g++-4.8
-            - libeigen3-dev
       env:
         - BUILD_TYPE='release'
         - CXX_COMPILER='g++-4.8'
+        - EIGEN_VER='3.3.4'
     - os: linux
       addons:
         apt:
           packages:
             - cmake3
             - cmake3-data
             - g++-4.8
-            - libeigen3-dev
       env:
         - BUILD_TYPE='release'
         - CXX_COMPILER='g++-4.8'
         - OPENMP='--omp'
+        - EIGEN_VER='3.3.4'
     - os: linux
       addons:
         apt:
           packages:
             - cmake3
             - cmake3-data
             - g++-4.8
-            - libeigen3-dev
             - lcov
       env:
         - BUILD_TYPE='debug'
         - CXX_COMPILER='g++-4.8'
         - COVERAGE='--coverage'
+        - EIGEN_VER='3.3.4'
 
 before_script:
+  - mkdir -p $HOME/Downloads/eigen
+  - cd $HOME/Downloads
+  - curl -Ls http://bitbucket.org/eigen/eigen/get/${EIGEN_VER}.tar.gz | tar -xz -C eigen --strip-components=1
+  - cd eigen
+  - cmake -H. -Bbuild_eigen -DCMAKE_INSTALL_PREFIX=$HOME/Deps/eigen &> /dev/null
+  - cmake --build build_eigen -- install &> /dev/null
   - cd ${TRAVIS_BUILD_DIR}
   - python --version
   - cmake --version
   - ${CXX_COMPILER} --version
+  - rm -rf $HOME/Downloads/eigen
```

Comment:
Instead of removing it every time you can use the [cache on Travis](https://docs.travis-ci.com/user/caching/#Arbitrary-directories). You won't spend time downloading and installing. Look also [here](https://github.com/bast/cmake-recipes/blob/master/testing/dependencies/install_dependencies.sh#L58-L73) and [here](https://github.com/bast/cmake-recipes/blob/master/.travis.yml#L59-L67)